### PR TITLE
Update python-app.yml

### PR DESCRIPTION
**Pull request by findyoky**
Added pip-audit for dependency check for auditing purpose.

### CODE REVIEW

File: .github/workflows/python-app.yml [hunk 1]
```diff
@@ -35,8 +35,12 @@ jobs:
         python -m pip install --upgrade pip
         pip install -r requirements.txt
         pip install -r requirements-dev.txt
+        pip install pip-audit==2.2.0
```
Comment (elveskevtar) [inline .github/workflows/python-app.yml:38]: I think we should add this to requirements-dev.txt

Comment (wbc2025) [reply .github/workflows/python-app.yml:38]: I agree that this should be changed

**Review by wbc2025 — CHANGES_REQUESTED** [review]
See Kevin's original comment

File: .github/workflows/python-app.yml (hunk 1 above)
Comment (findyoky) [reply .github/workflows/python-app.yml:38]: What is the file requirements-dev.txt used for?

Comment (wbc2025) [reply .github/workflows/python-app.yml:38]: It's used to put all the dependencies that a developer needs, but aren't needed to run to app in production. It allows the users to install the dev dependencies and run locally the kinds of checks that happen during the workflows. That way, a dev can perform the checks locally and fix any problems before pushing to the server.

**Comment by findyoky** [comment]
Canceled this change request.

**Pull request closed without merging**